runner.go

```diff
@@ -713,3 +716,77 @@ func newWatcher(c *Config, clients *dep.ClientSet, once bool) (*watch.Watcher, e
 	}
 	return w, nil
 }
+
+// applyConfigEnv applies custom env variables and whitelist/blacklist rules from config
+func (r *Runner) applyConfigEnv(env map[string]string) map[string]string {
+	// Parse custom environment variables
+	custom := make(map[string]string, len(r.config.Exec.Env.Custom))
+	for _, v := range r.config.Exec.Env.Custom {
+		list := strings.SplitN(v, "=", 2)
+		custom[list[0]] = list[1]
+	}
+
+	// In pristine mode, just return the custom environment. If the user did not
+	// specify a custom environment, just return the empty slice to force an
+	// empty environment. We cannot return nil here because the later call to
+	// os/exec will think we want to inherit the parent.
+	if config.BoolVal(r.config.Exec.Env.Pristine) {
+		if len(custom) > 0 {
+			return custom
+		}
+		return make(map[string]string)
+	}
+
+	keys := make(map[string]bool, len(env))
+	for k, _ := range env {
+		keys[k] = true
+	}
+
+	// anyGlobMatch is a helper function which checks if any of the given globs
+	// match the string.
+	anyGlobMatch := func(s string, patterns []string) bool {
+		for _, pattern := range patterns {
+			if matched, _ := filepath.Match(pattern, s); matched {
+				return true
+			}
+		}
+		return false
+	}
+
+	// Filter to envvars that match the whitelist
+	n := len(r.config.Exec.Env.Whitelist)
+	if n > 0 {
```

Comment:
I suggest changing this to scope the `n` variable to the block. Having it outside the block suggests it is used outside the block and it isn't. Eg.

`if n := len(r.config.Exec.Env.Whitelist); n > 0 {`